### DISCUSSION TITLE
Is there a way to always receive full payment instead of partial payments?

### DISCUSSION THREAD

**Discussion by olimits7** [btcpayserver/btcpayserver#4092] [Technical Support] [answered]
Hi,

I've been noticing when customers send me payments they have been getting sent a **paidPartial**, I'm guessing because of the network fee charged.  I have **Add additional fee (network fee) to invoice …on every payment** but it still generates a **paidPartial** invoice.

Is there a way to guarantee full invoice amount will get paid each time?

Thank you!

**Comment by olimits7** [comment]
I just tried testing again but it did **paidPartial** payment again.  I entered in $10.00 as the amount I wanted to deposit but now it's saying I still need to make another transaction.

https://gyazo.com/14e597b217c8ae4cdc04d8c4dcf8be2d

**Comment by olimits7** [comment, marked as answer]
I ended up using slippage setting in BTCPay Server and this seems to be a workaround for now.  I set slippage to 5% and it seems to work in receiving the paid full response.

**Reply by pavlenex** [reply]
It's not a workaround, you cannot enter 10$ in wallet and pay, scan the invocie and pay the exact amount that BTCPay asks for in bitcoin.

**Reply by olimits7** [reply]
Ok, so are you saying that I should add my own fee on top of the 10$ my customer is depositing?  For example, 10$+1$(fee) to avoid issues with partial payment with BTCPayServer?

**Reply by dennisreimann** [reply]
I think you are overcomplicating it.  Just pay the exact amount in bitcoin (or other selected cryptocurrency) instead of manually choosing a fiat based amount. The merchants charge bitcoin (or crypto), not fiat – so the denomination of the invoice isn't in fiat.

**Reply by olimits7** [reply]
That's what I did, I'm not entering in fiat on my **BitPay Wallet** when sending the payment; I'm entering the total due which was **0.00053600 BTC** but then after sending payment BTCPayServer said I was short **0.00000207 BTC**.

https://gyazo.com/d7c02479b9eb8f1339e2273145eb3033

I believe the issue is what @pavlenex stated below and when sending payments with these **BitPay/Coinbase Wallets** they're taking an additional fee, but that means every payment my customer sends using these wallets will be **underpaid** unless I add an additional fee of my own on top of the deposited amount so it at least is **overpaid** instead of **underpaid**.

**Comment by Kukks** [comment]
No, he is saying you should copy the bitcoin address and amount that you
need to send from the invoice and not use a $ amount on the customer wallet

**Reply by olimits7** [reply]
That's what I did, on this "**Awaiting Payment**" pop-up box on my website the original total amount due said **0.00053600 BTC** this was including the "**Network Cost**" that **BTCPay Server**" included too.  Then in my wallet I typed in the total due **0.00053600 BTC** and now it's saying I still need to make an additional payment of **0.00000207 BTC**.

https://gyazo.com/d7c02479b9eb8f1339e2273145eb3033

**Reply by pavlenex** [reply]
Which wallet do you use, if you're sending from an exchange, the exchange is taking the fee on your total, leading the underpaid invoice on BTCPay's end.

**Reply by olimits7** [reply]
I set up an **Electrum** wallet with my **Trezor** device using "**Legacy Segwit**" and this is the wallet I have added to my **BTCPayServer**.

To send the payment to my **BTCPayServer**, I used my **BitPay Wallet** and also did a test with my **CoinBase Wallet**; with same underpaid issue.  So you're saying **BitPay/Coinbase** are adding their own fee and that's why it keeps getting received as underpaid?

**Reply by astupidmoose** [reply]
> So you're saying **BitPay/Coinbase** are adding their own fee and that's why it keeps getting received as underpaid?

I'm not familiar with Bitpays wallet, but Coinbase definitely takes their own fee from the payment you send. Almost all custodial wallets do.

**Comment by olimits7** [comment]
I did some more research into this and it seems like the issue revolves around the "**BTCPayServer > Network Cost**" and because my "**BitPay Wallet**" cuts off the last 2 decimal places which is causing the variance amount.

**BTCPay Server "Awaiting Payment"**: https://gyazo.com/698dc699faf1880715837c48acbaac83
**BTCPay Server "Payment Made"**: https://gyazo.com/71816b9bb9a1ebae836e386807014854

**BitPay Wallet "Confirm Payment"**: https://gyazo.com/ca41ce798e91f6487d13a6f4e7d5726e
**BitPay Wallet "Payment Made"**: https://gyazo.com/a5ab4593f83fc49847ad7c7f369b3e6c

**Excel Analysis**: https://gyazo.com/d80efeb81c578e5d415c429a8611a28d

I'm going to do another test with "**Network Cost**" disabled in **BTCPay Server** and I'm sure I'll end up with an **under paid** invoice of **0.00000031 BTC** because for some reason "**BitPay Wallet**" is cutting off the last 2 decimal places.  I guess that's another issue I have to check with **BitPay** on.

**Comment by Kukks** [comment]
Yes, although if you are ok with those 2 decimal points being cut off, you
can add a payment tolerance percentage in your store's setting.

**Reply by olimits7** [reply]
Yeah, I guess that's the only way to resolve this issue; I set "**Consider the invoice paid even if the paid amount is ... % less than expected**" to **2%**.  

I did another test without adding network cost fees from **BTCPay Server** and it was **under paid** by just **0.00000057** the difference between the **6 decimal** places that **BitPay Wallet** uses versus the **8 decimal places** that **BTCPay Server** uses.

I wish there was a way to set the number of decimal places either in **BitPay Wallet** or **BTCPay Server** but it doesn't look like this is an option.

Ty.

**Reply by pavlenex** [reply]
BitPay sucks is moral of this story I think.

**Reply by olimits7** [reply]
> BitPay sucks is moral of this story I think.

Yes, that too! LOL
